infrastructure/src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -254,7 +254,11 @@ public synchronized void handleFailureDetectedMsg(CorfuPayloadMsg<FailureDetecto
         log.info("Received Failures : {}", msg.getPayload().getNodes());
         r.sendResponse(ctx, msg, new CorfuMsg(CorfuMsgType.ACK));
         FailureHandlerDispatcher failureHandlerDispatcher = new FailureHandlerDispatcher();
-        failureHandlerDispatcher.dispatchHandler(failureHandlerPolicy, latestLayout, getCorfuRuntime(), msg.getPayload().getNodes().keySet());
+        try {
+            failureHandlerDispatcher.dispatchHandler(failureHandlerPolicy, (Layout) latestLayout.clone(), getCorfuRuntime(), msg.getPayload().getNodes().keySet());
```

Comment:
Yes will change this naming in the next PR.
Want to keep this PR limited to seal changes.